package/kedro_viz/integrations/kedro/run_hooks.py

```diff
@@ -0,0 +1,249 @@
+            Kedro data catalog for size and metadata lookups.
+        _current_node (Optional[KedroNode]): Currently executing node, used for
+            error context and tracking.
+        _current_dataset (Optional[str]): Currently processed dataset name,
+            used for error context during I/O operations.
+        _current_operation (Optional[str]): Current dataset operation type
+            (e.g., 'loading', 'saving'), used for error context.
+        _all_nodes (list[KedroNode]): Complete list of nodes in the pipeline,
+            used for tracking execution progress.
+        _started_nodes (set[str]): Set of node names that have begun execution,
+            used to identify unstarted nodes during pipeline errors.
+    """
+
+    def __init__(self):
+        """
+        Initialize the pipeline run status hook.
+
+        Sets up empty tracking structures for monitoring pipeline execution,
+        including event collection, timing data, and error context management.
+        """
+        # Track times, events, and context for errors
+        self._node_start: dict[str, float] = {}
+        self._events: list[dict[str, Any]] = []
+        self._datasets: dict[str, Any] = {}
+        self._current_node: Optional[Any] = None
+        self._current_dataset: Optional[str] = None
+        self._current_operation: Optional[str] = None
+        self._all_nodes: list[KedroNode] = []
+        self._started_nodes: set[str] = set()
+
+    def _write_events(self) -> None:
+        """Persist events list to the project's .viz JSON file."""
+        write_events(self._events)
+
+    def _add_event(self, event: dict[str, Any], flush: bool = False) -> None:
+        """Append one event to the events list and optionally flush to disk."""
+
+        # We add events only for full/default pipeline as for MVP we only support
+        # full/default pipeline.
+        if len(self._all_nodes) == 0:
+            return
+
+        self._events.append(event)
+        if flush:
+            self._write_events()
+
+    def _set_context(self, dataset: str, operation: str, node: KedroNode) -> None:
+        """Save dataset I/O and node context for error handling."""
+        self._current_dataset = dataset
+        self._current_operation = operation
+        self._current_node = node
+
+    def _clear_context(self) -> None:
+        """Clear dataset I/O context after successful operations."""
+        self._current_dataset = None
+        self._current_operation = None
+
+    @hook_impl
+    def after_catalog_created(self, catalog: Any):
+        """
+        Triggered after the Kedro DataCatalog is constructed, before pipeline execution.
+        Captures references to all datasets for metadata lookups. At this point, no data
+        has been loaded or saved yet — so only static dataset configuration (not size/content)
+        can be accessed.
+        """
+        try:
+            # prefer new KedroDataCatalog
```

Comment:
Flagging this again. @rashidakanchwala this is not going to be in the Viz 11.1.0 right? so then I suggest to just remove it before we forget.